pkg/controller/crtconfigmap/crt_configmap_controller.go

```diff
@@ -0,0 +1,261 @@
+			Name: CrtConfigMapName,
+		},
+		Data: map[string]string{
+			"ca.crt": string(rootCA),
+		},
+	}
+
+}
+
+// NewCrtConfigMapController construct a new controller which would manage the configmap which stores
+// certificates in each namespace. It will make sure certificate configmap exists in each namespace.
+func NewCrtConfigMapController(cmInformer coreinformers.ConfigMapInformer, nsInformer coreinformers.NamespaceInformer, cl clientset.Interface, cm v1.ConfigMap) (*CrtConfigMapController, error) {
+	e := &CrtConfigMapController{
+		client:    cl,
+		configMap: cm,
+		queue:     workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "configmap"),
+	}
+	if cl != nil && cl.CoreV1().RESTClient().GetRateLimiter() != nil {
+		if err := metrics.RegisterMetricAndTrackRateLimiterUsage("crt_map_controller", cl.CoreV1().RESTClient().GetRateLimiter()); err != nil {
+			return nil, err
+		}
+	}
+
+	cmInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		DeleteFunc: e.configMapDeleted,
+		UpdateFunc: e.configMapUpdated,
+	})
+	e.cmLister = cmInformer.Lister()
+	e.cmListerSynced = cmInformer.Informer().HasSynced
+
+	nsInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    e.namespaceAdded,
+		UpdateFunc: e.namespaceUpdated,
+	})
+	e.nsLister = nsInformer.Lister()
+	e.nsListerSynced = nsInformer.Informer().HasSynced
+
+	e.syncHandler = e.syncNamespace
+
+	return e, nil
+
+}
+
+// CrtConfigMapController manages certificate ConfigMap objects inside Namespaces
+type CrtConfigMapController struct {
+	client    clientset.Interface
+	configMap v1.ConfigMap
+
+	// To allow injection for testing.
+	syncHandler func(key string) error
+
+	cmLister       corelisters.ConfigMapLister
+	cmListerSynced cache.InformerSynced
+
+	nsLister       corelisters.NamespaceLister
+	nsListerSynced cache.InformerSynced
+
+	queue workqueue.RateLimitingInterface
+}
+
+// Run starts process
+func (c *CrtConfigMapController) Run(workers int, stopCh <-chan struct{}) {
+	defer utilruntime.HandleCrash()
+	defer c.queue.ShutDown()
+
+	glog.Infof("Starting crt configmap controller")
+	defer glog.Infof("Shutting down crt configmap controller")
+
+	if !controller.WaitForCacheSync("crt configmap", stopCh, c.cmListerSynced, c.nsListerSynced) {
+		return
+	}
+
+	for i := 0; i < workers; i++ {
+		go wait.Until(c.runWorker, time.Second, stopCh)
+	}
+
+	<-stopCh
+}
+
+func (c *CrtConfigMapController) configMapDeleted(obj interface{}) {
+	if cm, err := convertToCM(obj); err != nil {
+		utilruntime.HandleError(err)
+	} else if cm.Name == CrtConfigMapName {
+		c.queue.Add(cm.Namespace)
+	}
+}
+
+func (c *CrtConfigMapController) configMapUpdated(oldObj, newObj interface{}) {
+	var (
+		oldConfigMap, newConfigMap *v1.ConfigMap
+		err                        error
+	)
+
+	if newConfigMap, err = convertToCM(newObj); err != nil {
+		utilruntime.HandleError(err)
+		return
+	}
+	if newConfigMap.Name != CrtConfigMapName {
+		return
```

Comment:
I don't think we should log a warning here since this could happen when user want to update any configmap besides ca configmap. 
Did you think of it as:
`if newConfigMap.Name != oldConfigMap.Name`